File: Backend/src/main/java/com/example/backend/entity/Candidate.java
package com.example.backend.entity;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "candidate")
public class Candidate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    // Other attributes
    private String name;
    private int score;
    @OneToMany(mappedBy = "candidate", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Result> results;
    @ManyToOne(fetch= FetchType.LAZY)
    @JoinColumn(name = "test_id")
    private Test test;


    // Constructors, getters, and setters
    // ...

    public Candidate() {
    }

    public Candidate(int id, String name, int score, List<Result> results, Test test) {
        this.id = id;
        this.name = name;
        this.score = score;
        this.results = results;
        this.test = test;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    public Test getTest() {
        return test;
    }

    public void setTest(Test test) {
        this.test = test;
    }
}

File: front_end/lib/app/core/models/candidate.dart
import 'result.dart';

class Candidate {
  final int id;
  final String name;
  final int score;
  final List<Result> results;
  final int testId;

  Candidate({
    this.id = -1,
    required this.name,
    this.score = 0,
    this.results = const [],
    this.testId = -1,
  });

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'name': name,
      'score': score,
      'results': results.map((x) => x.toJson()).toList(),
      'testId': testId,
    };
  }

  factory Candidate.fromJson(Map<String, dynamic> map) {
    return Candidate(
      id: map['id'] as int,
      name: map['name'] as String,
      score: map['score'] as int,
      results: map['results'] != null
          ? List<Result>.from(
              (map['results'] as List).map<Result>(
                (x) => Result.fromJson(x as Map<String, dynamic>),
              ),
            )
          : [],
      testId: map['testId'] as int,
    );
  }
}
